uui-core/src/data/processing/views/treeHelpers.ts

```diff
@@ -0,0 +1,73 @@
+import { IMap } from "../../../types";
+import { CompositeKeysMap } from "./CompositeKeysMap";
+
+export interface TreeParams<TItem, TId> {
+    getId?(item: TItem): TId;
+    getParentId?(item: TItem): TId | undefined;
+    complexIds?: boolean;
+}
+
+export function newMap<TKey, TValue>(params: TreeParams<any, any>) {
+    if (params.complexIds) {
+        return new CompositeKeysMap<TKey, TValue>();
+    } else {
+        return new Map<TKey, TValue>();
+    }
+}
+
+export const deleteFromList = <TId>(id: TId, list: TId[]) => {
+    const foundIndex = list.findIndex((childId) => childId === id);
+    if (foundIndex !== -1) {
+        list.splice(foundIndex, 1);
+    }
+
+    return list;
+};
+
+const pasteNewItemIntoList = <TItem, TId>(
+    id: TId,
+    newItem: TItem,
+    byId: IMap<TId, TItem>,
+    list: TId[],
+    comparator: (newItem: TItem, existingItem: TItem) => number,
+) => {
+    const listWithNewItem: TId[] = [];
+    list.forEach((itemId) => {
```

Comment:
Can we simplify this, and avoid o(n^2)? Probably we can
1. find the first index, where newItem is greater than existing (by comparator)
2. insert item into this position